app/views/job_invocations/_form.html.erb

```diff
@@ -0,0 +1,58 @@
+<%= form_for @composer.job_invocation, :html => { 'data-refresh-url' => refresh_job_invocations_path, :id => 'job_invocation_form' } do |f| %>
+  Assign host:
+  <%= @composer.targeted_hosts_count %>
+  <%= button_tag(:type => 'button', :class => 'btn btn-default btn-sm', :title => _("Refresh"), :id => 'refresh_execution_form') do %>
+    <%= icon_text('refresh') %>
+  <% end %>
+  <%= fields_for @composer.targeting do |t| %>
+    <%= t.hidden_field :bookmark_id, :value => params[:bookmark_id] %>
+    <%= textarea_f t, :search_query, :value => @composer.displayed_search_query, :rows => 5 %>
+    <%= radio_button_f t, :targeting_type, :value => Targeting::STATIC_TYPE, :text => _(Targeting::TYPES[Targeting::STATIC_TYPE]) %>
+    <%= radio_button_f t, :targeting_type, :value => Targeting::DYNAMIC_TYPE, :text => _(Targeting::TYPES[Targeting::DYNAMIC_TYPE]) %>
+  <% end %>
+
+  <hr >
+
+  <%= selectable_f f, :job_name, @composer.available_job_names %>
+
+  <% @composer.displayed_provider_types.each do |provider_type| %>
+    <fieldset id="provider_<%= provider_type %>" class="provider_form">
+      <% if @composer.needs_provider_type_selection? %>
+        <legend><%= _('Provider') + ': ' + provider_type %></legend>
+      <% end %>
+      <%= f.fields_for 'providers' do |p| %>
```

Comment:
![i-am-not-a-big-fan_o_25018](https://cloud.githubusercontent.com/assets/190443/9080296/8be50470-3b52-11e5-93ce-9e654f9f038a.jpg) of the 1-2 long vars, although the `f` is quite common, but `pt`, `jt`,`j`: it's getting a bit unreadable.